### PR TITLE
Exp backoff retry for discover services

### PR DESCRIPTION
**Pull request by tilak-puli**

### CODE REVIEW

File: android/src/main/java/io/mosip/tuvali/ble/central/impl/GattClient.kt [hunk 1]
```diff
@@ -117,7 +117,10 @@ class GattClient(var context: Context) {
       }
 
       Log.i(logTag, "discovered services: ${gatt?.services?.map { it.uuid }}")
-      onServicesDiscovered()
+      val services = gatt?.services?.map { it.uuid }
+      if (services != null) {
+        onServicesDiscovered(services)
+      }
```
Comment (tilak-puli) [inline android/src/main/java/io/mosip/tuvali/ble/central/impl/GattClient.kt:132]: handle else case

File: android/src/main/java/io/mosip/tuvali/ble/central/impl/Controller.kt [hunk 2]
```diff
@@ -156,8 +156,8 @@ class Controller(val context: Context) {
     messageSender.sendMessage(scanStartFailureMessage)
   }
 
-  private fun onServicesDiscovered(){
-    messageSender.sendMessage(DiscoverServicesSuccessMessage())
+  private fun onServicesDiscovered(services: List<UUID>){
```
Comment (tilak-puli) [inline android/src/main/java/io/mosip/tuvali/ble/central/impl/Controller.kt]: services feels like List<GattService>. Pls update var name

File: android/src/main/java/io/mosip/tuvali/ble/central/state/StateHandler.kt [hunk 3]
```diff
@@ -234,4 +235,24 @@ class StateHandler(
...
+
+////  override fun servicesDiscovered(msg: IMessage, services : List<UUID>) {
+////    val message = this.obtainMessage()
+////    message.what = msg.commandType.ordinal
+////    message.obj = msg
+////    val isSent = this.servicesDiscovered(message,services)
+////    if (!isSent) {
+////      Log.e(logTag, "sendMessageDelayed to state handler for ${msg.commandType} failed")
+////    }
+//  }
```
Comment (tilak-puli) [inline android/src/main/java/io/mosip/tuvali/ble/central/state/StateHandler.kt]: pls remove dead code

File: android/src/main/java/io/mosip/tuvali/common/retrymechanism/BackOffStrategy.kt [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        return waitTime
+    }
+
+    private fun calculateWaitTime(retryCount: Int): Double {
+        println(
+            "Retry count $retryCount and wait ${
+                initialWaitTime * multiplier.toDouble().pow(retryCount.toDouble())
+            }"
+        )
+        return initialWaitTime * multiplier.toDouble().pow(retryCount.toDouble())
```
Comment (krishnakumar4a4) [inline android/src/main/java/io/mosip/tuvali/common/retrymechanism/BackOffStrategy.kt]: Use an exponent other than retryCount in the power to multiplier

**Pull request merged**